distributor-node/src/services/networking/query-node/api.ts

```diff
@@ -172,12 +181,53 @@ export class QueryNodeApi {
     })
   }
 
-  public getDataObjectDetails(objectId: string): Promise<DataObjectDetailsFragment | null> {
-    return this.uniqueEntityQuery<GetDataObjectDetailsQuery, GetDataObjectDetailsQueryVariables>(
+  protected readFragment<FragmentT, VariablesT extends Record<string, unknown>>(
+    id: string,
+    fragment: DocumentNode
+  ): FragmentT | null {
+    // Get the fragment name, usually first element of the definitions array is the name of the top level fragment
+    const fragmentName = (fragment.definitions[0] as FragmentDefinitionNode).name.value
+    if (!fragmentName) {
```

Comment:
So I checked that `Id` isn't necessarily same as the fragment name. `id` refers to the cached object ID (e.g. for the `StorageDataObject` entity type it can be `StorageDataObject:1`) however the fragment name refers to the specific fragment of the entity type that we want to query from the cache, e.g. [DataObjectDetails](https://github.com/zeeshanakram3/joystream/blob/argus_improve_head_request_latency/distributor-node/src/services/networking/query-node/queries/queries.graphql#L23)